starboard/shared/win32/posix_emu/include/_remove_problematic_windows_macros.h

```diff
@@ -0,0 +1,22 @@
+// NOLINT(build/header_guard)
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file should be included in other include files within posix_emu, after
+// including MSVC headers that may define macros that collide with Cobalt code.
+// When undefining a macro here, provide a comment of which specific Windows
+// header file defined it for future reference.
+#undef NO_ERROR        // winerror.h; b/302733082#comment15
+#undef PostMessage     // winuser.h
```

Comment:
I think we get a pretty decent list from https://github.com/youtube/cobalt/blob/main/base/win/windows_types.h#L234-L257. I don't know that we want to add all of these or add them as we run into them, but good to know about.

I've at least run into DeleteFile/DeleteFileW before and had to change headers around to fix that

Reply:
Sounds good, then could you just add DeleteFile here too?